cmd/rook/ceph/sanitize_disk.go

```diff
@@ -0,0 +1,219 @@
+		go s.executeCommand(&wg)
+	}
+
+	wg.Wait()
+}
+
+func (s *DiskSanitizer) sanitizeLVMDisk(osdLVMList []oposd.OSDInfo) {
+	// Initialize work group to wait for completion of all the go routine
+	var wg sync.WaitGroup
+	pvs := []string{}
+
+	for _, osd := range osdLVMList {
+		// Increment the wait group counter
+		wg.Add(1)
+
+		// Assign disk
+		s.disk = osd.BlockPath
+
+		// Lookup the PV associated to the LV
+		pvs = append(pvs, s.returnPVDevice()[0])
+
+		// run c-v
+		go s.wipeLVM(osd.ID, &wg)
+	}
+	// Wait for ceph-volume to finish before wiping the remaining Physical Volume data
+	wg.Wait()
+
+	var wg2 sync.WaitGroup
+	// // purge remaining LVM2 metadata from PV
+	for _, pv := range pvs {
+		wg2.Add(1)
+		s.disk = pv
+		go s.executeCommand(&wg2)
+	}
+	wg2.Wait()
+}
+
+func (s *DiskSanitizer) wipeLVM(osdID int, wg *sync.WaitGroup) {
+	// On return, notify the WaitGroup that we’re done
+	defer wg.Done()
+
+	output, err := s.context.Executor.ExecuteCommandWithCombinedOutput("stdbuf", "-oL", "ceph-volume", "lvm", "zap", "--osd-id", strconv.Itoa(osdID), "--destroy")
+	if err != nil {
+		logger.Errorf("failed to sanitize osd %d. %s. %v", osdID, output, err)
+	}
+
+	logger.Infof("%s\n", output)
+	logger.Infof("successfully sanitized lvm osd %d disk %q", osdID, s.disk)
+}
+
+func (s *DiskSanitizer) returnPVDevice() []string {
+	logger.Infof("s.disk: %s", s.disk)
+
+	output, err := s.context.Executor.ExecuteCommandWithOutput("lvs", s.disk, "-o", "seg_pe_ranges", "--noheadings")
+	if err != nil {
+		logger.Errorf("failed to execute lvs command. %v", err)
+		return []string{}
+	}
+
+	logger.Infof("output: %s", output)
+	return strings.Split(output, ":")
+}
+
+func (s *DiskSanitizer) buildDDArgs() []string {
+	ddArgs := []string{
+		s.sanitizeDiskSpec.SanitizeUtility.String(),
+		fmt.Sprintf("if=%s", s.buildDataSource()),
+		fmt.Sprintf("of=%s", s.disk),
+		fmt.Sprintf("bs=%s", ddBS),
+		fmt.Sprintf("oflag=%s", ddFlags),
+	}
+
+	switch s.sanitizeDiskSpec.SanitizeMethod {
+	case cephv1.SanitizeMethodMetadata:
+		ddArgs = append(ddArgs, fmt.Sprintf("count=%s", ddCount))
+	}
+
+	return ddArgs
+}
+
+func (s *DiskSanitizer) buildShredArgs() []string {
+	return []string{
+		s.sanitizeDiskSpec.SanitizeUtility.String(),
+		"--force",
+		"--zero",
+		"--verbose",
+		fmt.Sprintf("--random-source=%s", s.buildDataSource()),
+		fmt.Sprintf("--iterations=%s", strconv.Itoa(int(sanitizeIteration))),
+		s.disk}
+}
+
+func (s *DiskSanitizer) buildDataSource() string {
+	return fmt.Sprintf("/dev/%s", s.sanitizeDiskSpec.SanitizeDataSource.String())
+}
+
+func (s *DiskSanitizer) executeCommand(wg *sync.WaitGroup) {
+	// On return, notify the WaitGroup that we’re done
+	defer wg.Done()
+
+	// Empty arg array
+	args := []string{}
+
+	switch s.sanitizeDiskSpec.SanitizeUtility {
+	case cephv1.SanitizeUtilityDD:
+		args = s.buildDDArgs()
+	case cephv1.SanitizeUtilityShred:
+		args = s.buildShredArgs()
+	}
+
+	// Base command to run
+	baseCommand := "stdbuf"
```

Comment:
The `stdbuf -oL -eL` I thought would only be needed when running a python app like c-v. Here we are just running dd or shred, right?

Reply:
Yes, left-over attempt for `shred` to print its progress... Not needed, thanks.